src/errors/errors.ts

```diff
@@ -295,9 +300,16 @@ export type ContractWriteErrorParams<ContractFunctionName> = Omit<
     "contractErrorCode"
 >
 
+export class CollateralDepositCapError extends ContractWriteError<Vault> {
+    constructor(data: ContractWriteErrorParams<keyof Vault>) {
+        super({ ...data })
+        this.name = ErrorName.COLLATERAL_DEPOSIT_CAP_ERROR
+    }
+}
+
 export class PriceSlippageCheckError extends ContractWriteError<ClearingHouse> {
     constructor(data: ContractWriteErrorParams<keyof ClearingHouse>) {
-        super({ ...data, contractErrorCode: ContractErrorCode.PRICE_SLIPPAGE_CHECK_FAILS_PSCF })
+        super({ ...data })
```

Comment:
Should we remove all hardcoded `contractErrorCode` param in other error constructors?